packages/sync/src/offline/OfflineRestoreHandler.ts

```diff
@@ -41,28 +42,28 @@ export class OfflineRestoreHandler {
     // if there is no offline data  then just exit
     if (!this.hasOfflineData()) { return; }
 
+    // wait before it was cleared
+    await this.clearOfflineData();
+
     logger("Replying offline mutations after application restart");
-    // return as promise, but in the end clear the storage
-    const uncommittedOfflineMutation: OperationQueueEntry[] = [];
 
-    await Promise.all(this.offlineData.map(async (item) => {
-      try {
-        await this.apolloClient.mutate({
-          variables: item.operation.variables,
-          mutation: item.operation.query,
-          context: item.operation.getContext
-        });
-      } catch (e) {
-        // set the errored mutation to the stash
-        uncommittedOfflineMutation.push(item);
-      }
-    }));
+    this.offlineData.forEach(item => {
+      this.apolloClient.mutate({
+        variables: item.operation.variables,
+        mutation: item.operation.query,
+        optimisticResponse: item.optimisticResponse
+      });
+    });
 
-    // wait before it was cleared
-    await this.clearOfflineData();
+    await this.waitForOperationsEnqueued();
```

Comment:
this works but I think we can simply use listener for enqueued operations and await for this.

Reply:
Was also thinking about that, but not really sure how to do that - user provides his own listener via `clientConfig`. So should we wrap it, so that we can listen for events as well?